transforms/helpers/util/get-telemetry-for.js

```diff
@@ -23,27 +38,63 @@ for (let packagePath of packagePaths) {
  * @param {String} filePath the path on disk (from current working directory)
  * @returns {String} The in-browser module path for the specified filePath
  */
-function getModulePathFor(filePath, addonPaths = ADDON_PATHS) {
-  let fileSegments = filePath.split('/');
-  let addonSegments = [];
+function getModulePathFor(filePath, addonPaths = ADDON_PATHS, appPaths = APP_PATHS) {
+  let bestMatch = '';
+  let moduleNameRoot, relativePath, isApp, result;
 
-  while (fileSegments.length > 0) {
-    addonSegments.push(fileSegments.shift());
+  for (let addonPath in addonPaths) {
+    if (filePath.startsWith(addonPath) && addonPath.length > bestMatch.length) {
+      bestMatch = addonPath;
+      moduleNameRoot = addonPaths[addonPath];
+      relativePath = filePath.slice(
+        addonPath.length + 1 /* for slash */,
+        -path.extname(filePath).length
+      );
+    }
+  }
 
-    if (addonPaths[addonSegments.join('/')]) {
-      break;
+  for (let appPath in appPaths) {
+    if (filePath.startsWith(appPath) && appPath.length > bestMatch.length) {
+      bestMatch = appPath;
+      moduleNameRoot = appPaths[appPath];
+      relativePath = filePath.slice(
+        appPath.length + 1 /* for slash */,
+        -path.extname(filePath).length
+      );
+      isApp = true;
     }
   }
 
-  let addonFilePath = addonSegments.join('/');
-  let addonName = addonPaths[addonFilePath];
+  if (!relativePath && process.cwd() === path.resolve(__dirname, '../../..')) {
+    // this is pretty odd, but our tests in
+    // transforms/ember-object/__testfixtures__ don't actually live in an ember
+    // app or addon, so the standard logic above doesn't work for them
+    //
+    // this works by passing through the input file name when we are operating
+    // on the local ember-es6-class-codemod repo **and** we were not able to
+    // resolve a relativePath via normal means
+    return filePath.replace(/\.[^/.]+$/, '');
+  }
+
+  if (!relativePath) {
+    return;
+  }
 
-  let relativeFilePath = fileSegments
-    .join('/')
-    .replace(/^(addon|app)\//, '')
-    .replace(/\.[^/.]+$/, '');
+  if (isApp) {
+    if (relativePath.startsWith('app')) {
+      result = `${moduleNameRoot}${relativePath.slice(3)}`;
+    } else if (relativePath.startsWith('tests')) {
+      result = `${moduleNameRoot}/${relativePath}`;
+    }
+  } else {
+    if (relativePath.startsWith('addon-test-support')) {
```

Comment:
I believe addons could have files in an `app/` folder, no? Also, wouldn't this ignore test files in addons?

Reply:
Ah, I guess addon files in the app folder are inaccessible anyways - no way to tell if they were part of the app, or the addon 😕

Reply:
@pzuraq addon's with files in the `app/` folder are super problematic for us, they "appear" as if they are in the app itself (from a module name perspective).